### PR TITLE
Add possibility to save IR in transform.py

### PR DESCRIPTION
**Pull request by giuseros**
cc @nicolasvasilache

### CODE REVIEW

File: python/examples/core/transform.py [hunk 1]
```diff
@@ -209,6 +239,25 @@ def print_ir(self,
...
+  def save_ir(self,
+              file_name,
+              after_all: bool = False,
+              llvm: bool = False) -> TransformationList:
+    """Return a new transformation list that save the IR at the given points."""
+    transforms = []
+    id = 0
+    for t in self.transforms:
+      if (len(transforms) >= 2 and
+          type(transforms[-2]).__name__ == type(t).__name__):
```
Comment (nicolasvasilache) [inline python/examples/core/transform.py:251]: Do you see value in this type of granularity?
I would have thought that set by step saving and incrementing would be simpler and nicer.

I.e. atm you'd get something like:
```
foo.transA.0
foo.transA.1
foo.transB.0
foo.transB.1
```
which does not tell you how A and B are interleaved.

How about 
```
foo.1.transA.
foo.2.transB
foo.3.transA
foo.4.transB
```

instead ?

File: python/examples/core/transform.py [hunk 2]
```diff
@@ -134,6 +134,36 @@ def __call__(self, module: Module, fun_name: str):
     return module
 
 
+class SaveIR(Transform):
```
Comment (nicolasvasilache) [inline python/examples/core/transform.py:137]: Note that the codegen strategy pipeline will soon be deprecated in favor of the scheduling dialect.
@ftynse may prefer you hold off on this?

Comment (giuseros) [reply python/examples/core/transform.py:137]: Well if it is really a problem I can try to move it in the experimental/ folder as well. Please, let me know!

Comment (nicolasvasilache) [reply python/examples/core/transform.py:137]: I think it is really about reducing extra work for @ftynse he should say what he prefers.

**Review by ftynse — COMMENTED** [review]
The addition is not problematic for me in itself.

File: python/examples/core/transform.py (hunk 1 above)
Comment (nicolasvasilache) [reply python/examples/core/transform.py:251]: coming back to this, I don't think this has been addressed ?

**Comment by nicolasvasilache** [comment]
> The addition is not problematic for me in itself.

Ok then let's address the outstanding comment and land this please!

File: python/examples/core/transform.py (hunk 1 above)
Comment (giuseros) [reply python/examples/core/transform.py:251]: No, sorry, I didn't notice @ftynse comment. I am going to push it now

Comment (giuseros) [reply python/examples/core/transform.py]: Done :)

**Review by nicolasvasilache — APPROVED** [review]

**Pull request merged**